nautobot/extras/api/views.py

```diff
@@ -77,12 +78,37 @@ def get_view_name(self):
         return "Extras"
 
 
+class NotesViewSetMixin(object):
+    @action(detail=True, url_path="notes", methods=["get", "post"])
+    def notes(self, request, pk=None):
+        """
+        API methods for returning or creating notes on an object.
+        """
+        obj = get_object_or_404(self.queryset, pk=pk)
+        if request.method == "POST":
+            content_type = ContentType.objects.get_for_model(obj)
+            data = request.data
+            data["assigned_object_id"] = obj.pk
+            data["assigned_object_type"] = f"{content_type.app_label}.{content_type.model}"
+            serializer = serializers.NoteSerializer(data=data, context={"request": request})
+
+            # Create the new Note.
+            serializer.is_valid(raise_exception=True)
+            serializer.save(user=request.user)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        else:
+            serializer = serializers.NoteSerializer(obj.notes, many=True, context={"request": request})
```

Comment:
Do we need to do anything to support pagination here? Or is it auto-magic?

Reply:
Good question, I thought pagination was auto-magic. Also, not really using the NestedNoteSerializer anywhere, since we are instead rewriting to a URL. But the NestedNoteSerializer is required due to Customfields and such.